homeassistant/components/roborock/__init__.py

```diff
@@ -43,19 +47,31 @@ async def async_setup_entry(hass: HomeAssistant, entry: ConfigEntry) -> bool:
         for device in device_map.values()
     }
     network_results = await asyncio.gather(
-        *(mqtt_client.get_networking() for mqtt_client in mqtt_clients.values())
+        *(mqtt_client.get_networking() for mqtt_client in mqtt_clients.values()),
```

Comment:
From the [pyhton docs](https://docs.python.org/3/library/asyncio-task.html#asyncio.gather)

> If return_exceptions is False (default), the first raised exception is immediately propagated to the task that awaits on gather(). Other awaitables in the aws sequence won’t be cancelled and will continue to run.

Before this change, the setup failed if one client raised an error.
After this change, for each client, that raised an error, a log entry is created, but only if all clients failed, the setup is failing.

I don't have a roborock and not sure if my use case can really happen...
Example:
- I have two roborock vacuums set up correctly.
- I switch off one completely
- Restart HA
- I'm wondering why only one device was set up. Maybe it would be better to mark the unavailable one as unavailable instead of not creating the entities.

Reply:
I kind of went down a rabbit hole implementing this.

The ability to mark a coordinator as available or not and then being able to recover later on down the line seemed like a very good things to have.

So I reworked how a lot of __init__ (and files down the line) worked